Wikipedia/Code/WMFTableOfContentsAnimator.swift

```diff
@@ -182,16 +189,16 @@ public class WMFTableOfContentsAnimator: UIPercentDrivenInteractiveTransition, U
             self.presentingViewController?.presentViewController(self.presentedViewController!, animated: true, completion: nil)
         case (.Changed):
             let position = gesture.locationInView(gesture.view);
-            let distanceFromRight = CGRectGetMaxX(gesture.view!.bounds) - position.x
-            let transitionProgress = distanceFromRight / CGRectGetMaxX(gesture.view!.bounds)
+            let distanceFromSide = WMFTableOfContentsAnimator.shouldMoveTOCToLeft() ? position.x : CGRectGetMaxX(gesture.view!.bounds) - position.x
+            let transitionProgress = distanceFromSide / CGRectGetMaxX(gesture.view!.bounds)
             self.updateInteractiveTransition(transitionProgress)
         case (.Ended):
             self.isInteractive = false
-            let velocityRequiredToPresent = -CGRectGetWidth(gesture.view!.bounds)
+            let velocityRequiredToPresent = -CGRectGetWidth(gesture.view!.bounds) * rtlMultiplier()
             let velocityRequiredToDismiss = CGRectGetWidth(gesture.view!.bounds)
             
             let velocityX = gesture.velocityInView(gesture.view).x
-            if velocityX > velocityRequiredToDismiss{
+            if velocityX * rtlMultiplier() > velocityRequiredToDismiss{
```

Comment:
Tested and it appears correct. This case is only fallen into if you (LTR) begin dragging the TOC in from the right edge, then quickly flick it *back toward* the right edge, or if RTL begin dragging the TOC in from the left edge then quickly flick it back toward the left edge. The existing code handles both cases in the same logical way as the old code did for LTR.